### DISCUSSION TITLE
Service restart command

### DISCUSSION THREAD

**Discussion by Gijeeg86** [DNSCrypt/dnscrypt-proxy#2606] [Q&A]
Didn’t find an exact answer in the wiki. Let’s say I have set a cron for a script to update every day a blocklist external txt file. Do I need to launch the dnscrypt-proxy service restart command as well? In other words if I change the content of the blocklist.txt do I need to restart the dnscrypt-proxy?

**Comment by lifenjoiner** [comment]
Yes, you do. Try then you will know.

**Reply by Gijeeg86** [reply]
I was asking since I have this strange behavior doing following trials:

1) I stop the service
2) I add to the downloaded blocklist file a line with *.youtube.*
3) I start again the service
4) both Safari and Firefox block youtube.com
So far so good...
Then I remove the *.youtube.* line from the file (in between stop and start service commands) and both Safari and Firefox let me communicate with youtube.com
So far so good...
now if I modify again the blocklist file putting again the entry for youtube.* (in between stop and start) both Safari and Firefox are NOT blocking youtube.com and to make the changes effective I have to reboot the OS.
What am I missing?

**Reply by lifenjoiner** [reply]
There is no block/allow lists reloading at all in the program.
There must be something special on your side, I guess.

And, I guess you are saying `*.youtube.*` rather than `.youtube.`.

**Reply by Gijeeg86** [reply]
Yes I mean with “*”

**Reply by Gijeeg86** [reply]
Honestly I don't know what's wrong. NSLOOKUP command is reflecting the blocklist file  "real-time" changes after restarting the proxy....the browsers don't. I have also flushed the DNS cache of both Safari and Firefox but realtime changes on the blocklist txt files are not caught as I described above

**Reply by lifenjoiner** [reply]
Long connection reused? Did you restart the browser?

**Reply by Gijeeg86** [reply]
I think I found the root cause. On MacOS the mDNSResponder daemon is the system-wide DNS resolver. From this manual definition I desume that it acts as "hub" for all the DNS queries started from the various apps...so flushing mDNSResponder cache (sudo killall -HUP mDNSResponder) after restarting the dnscrypt proxy it seems to solve the issue. I'm still testing it
